### PR TITLE
Update stories submodule reference

### PR DESCRIPTION
**Pull request by aforcier**
Updates the submodule reference for Stories to bring in the latest fixes.

Includes:
https://github.com/Automattic/stories-android/pull/603
https://github.com/Automattic/stories-android/pull/605
https://github.com/Automattic/stories-android/pull/608
https://github.com/Automattic/stories-android/pull/609

@loremattei I'm targeting 16.2 with this branch, but it should only be merged if https://github.com/wordpress-mobile/WordPress-Android/pull/13393 has been merged in time for the release cut, and reinstates the Stories+mobile-gb work that was reverted in https://github.com/wordpress-mobile/WordPress-Android/pull/13392.

### To test:
Basically just check that you're able to launch the story creator (for a WP.com site, My site or post list FAB > Story post > Select media > Enter story editor).

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by mzorz — APPROVED** [review]
🚂 all the way!
:shipit:
